---
title: "Single-marker MLM GWAS using GAPIT"
author: "Haipeng Yu & Gota Morota"
date: "April 1, 2020"
output: 
    prettydoc::html_pretty:
      theme: architect
      highlight: vignette
      toc: true
---
  
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
## Overview

This example illustrates fitting a single marker GWAS model using GAPIT. The detailed documentation of GAPIT can be found at [http://zzlab.net/GAPIT/](http://zzlab.net/GAPIT/)

## Load packages 

```{r, eval = FALSE}
# clear working environment
rm(list=ls()) 

# install and load support packages
source("http://www.zzlab.net/GAPIT/GAPIT.library.R") 

# load GAPIT function
source("http://www.zzlab.net/GAPIT/gapit_functions.txt")
```


## Maize data

The maize data set used here is from a [maize association panel](https://www.panzea.org/) including 281 diverse lines genotyped with 3,093 markers. The three phenotypes include ear height (EarHT), days to pollination (dpoll), and ear diameter (EarDia). In this example, we will only use EarHT. 

```{r, eval = FALSE}
# phenotypes
myY <- read.table(file = "http://zzlab.net/GAPIT/data/mdp_traits.txt", header = TRUE)

# marker matrix
myGD <- read.table(file = "http://zzlab.net/GAPIT/data/mdp_numeric.txt", header = TRUE)

# map information of markers 
myGM <- read.table(file = "http://zzlab.net/GAPIT/data/mdp_SNP_information.txt", header = TRUE)
```


## Mixed linear model (MLM) GWAS
In this example, we will perform single marker GWAS analysis using a mixed linear model including the first three principle components (PCs) and the genomic relationship matrix. We will estimate the variance components only once prior to GWAS by setting `SNP.3D = TRUE`. 

```{r, eval = FALSE}
# GWAS with MLM 
gwasMLMfit <- GAPIT(Y = myY[, c(1:2)], GD = myGD, GM = myGM, model = 'MLM', 
                    PCA.total = 3, SNP.P3D = TRUE, SNP.MAF = 0.05)
```

## Results
Use the `read.csv()` function to read the output file `GAPIT.MLM.EarHT.GWAS.Results.csv`.  The columns 4, 9 and 10 are p-value, FDR adjusted p-value, and marker effect, respectively.  
```{r}
gwasMLM <- read.csv('GAPIT.MLM.EarHT.GWAS.Results.csv', header = TRUE)
head(gwasMLM)
```

## Manhattan plot
Open the `GAPIT.MLM.EarHT.Manhattan.Plot.Genomewise.pdf` to view the Manhattan plot. 
<img src="GAPIT.MLM.EarHT.Manhattan.Plot.Genomewise.pdf" alt="GAPIT.MLM.EarHT.Manhattan.Plot.Genomewise.pdf">



